Simple Inventory Management System/Program.cs

```diff
@@ -1,86 +1,102 @@
-                        Console.WriteLine("Enter the name of the product to delete: ");
-                        string productToDelete = Console.ReadLine();
-                        inventory.DeleteProduct(productToDelete);
+                    case 3:
+                        // Edit Product
+                        Console.WriteLine("Enter the name of the product to be updated:");
+                        string productNameToUpdate = Console.ReadLine();
+                        Console.WriteLine("Enter the new name:");
+                        string newName = Console.ReadLine();
+                        Console.WriteLine("Enter the new price:");
+                        decimal newPrice = decimal.Parse(Console.ReadLine());
+                        Console.WriteLine("Enter the new quantity:");
+                        int newQuantity = int.Parse(Console.ReadLine());
+
+                        await productRepository.EditProductAsync(productNameToUpdate, newName, newPrice, newQuantity);
                         break;
 
-                    case "5":
-                        Console.WriteLine("Enter the name of the product to search: ");
-                        string productToSearch = Console.ReadLine();
-                        inventory.SearchProduct(productToSearch);
+                    case 4:
+                        // Delete Product by ID
+                        Console.WriteLine("Enter the ID of the product to be deleted:");
+                        int productIdToDelete;
+                        while (!int.TryParse(Console.ReadLine(), out productIdToDelete) || productIdToDelete < 0)
+                        {
+                            Console.WriteLine("Invalid ID. Please enter a positive number.");
+                        }
+
+                        await productRepository.DeleteProductAsync(productIdToDelete);
                         break;
 
-                    case "6":
-                        Console.ForegroundColor = ConsoleColor.Magenta;
-                        Console.WriteLine("Exiting...");
-                        Console.ForegroundColor = ConsoleColor.White;
-                        return;
 
+                    case 5:
+                        // Search Product
+                        Console.WriteLine("Enter the name of the product to search:");
+                        string productNameToSearch = Console.ReadLine();
+
+                        await productRepository.SearchProductAsync(productNameToSearch);
+                        break;
 
                     default:
-                        Console.WriteLine("Invalid choice. Please try again.");
+                        Console.WriteLine("Please enter a valid choice");
                         break;
                 }
-            }
+                Console.WriteLine("\nDo you want to continue? (Yes/No)");
+                answer = Console.ReadLine();
+            } while (answer.ToUpper() == "YES" || answer.ToUpper() == "Y");
```

Comment:
I don't remember when, but I remember that I have told you about using comparison like:
`string.Equals(val, "YES", StringComparison.OrdinalIgnoreCase)`